src/utils.ts

```diff
@@ -292,7 +321,15 @@ export function handleTokens(token: string): Token {
 					secret: p.s,
 					C: bytesToHex(p.c),
 					amount: p.a,
-					id: bytesToHex(t.i)
+					id: bytesToHex(t.i),
+					dleq:
```

Comment:
We can avoid `dleq:undefined` being present on every token without dleqs by doing something like

```ts
proofs.push({
    secret: p.s,
    C: bytesToHex(p.c),
    amount: p.a,
    id: bytesToHex(t.i),
    ...(p.d && {
        dleq: {
	    e: bytesToHex(p.d.e),
	    s: bytesToHex(p.d.s),
	    r: bytesToHex(p.d.r)
	} as SerializedDLEQ
    })
});
```

What do you think?

Reply:
I tried using this elegant solution in the place of other ugly `dleq == undefined ? undefined : ...` check it out.